velox/serializers/tests/UnsafeRowSerializerTest.cpp

```diff
@@ -0,0 +1,131 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include "velox/serializers/UnsafeRowSerializer.h"
+#include <gtest/gtest.h>
+#include "velox/vector/fuzzer/VectorFuzzer.h"
+#include "velox/vector/tests/VectorTestBase.h"
+
+using namespace facebook::velox;
+
+class UnsafeRowSerializerTest : public ::testing::Test {
+ protected:
+  void SetUp() override {
+    pool_ = memory::getDefaultScopedMemoryPool();
+    serde_ = std::make_unique<serializer::spark::UnsafeRowVectorSerde>();
+  }
+
+  void sanityCheckEstimateSerializedSize(
+      RowVectorPtr rowVector,
+      const folly::Range<const IndexRange*>& ranges) {
+    auto numRows = rowVector->size();
+    std::vector<vector_size_t> rowSizes(numRows, 0);
+    std::vector<vector_size_t*> rawRowSizes(numRows);
+    for (auto i = 0; i < numRows; i++) {
+      rawRowSizes[i] = &rowSizes[i];
+    }
+    serde_->estimateSerializedSize(rowVector, ranges, rawRowSizes.data());
```

Comment:
```rawRowSizes[i] = &rowSizes[i];``` seems to be setting ```rawRowSizes``` all to zero. How this function sanity checks the estimated size.